beets-bandcamp: init at 2022-06-07

rrix opened the pull request:
###### Description of changes

This adds [beetcamp](https://github.com/snejus/beetcamp) and makes it available for the `pluginOverrides` as `beetsPackages.beetcamp`. There is a `beetcamp` CLI included which would be nice to expose if the package is included directly in a profile but propagating beets-minimal in this plugin would break the beets-maximal this plugin is eventually included within... 

Happy to take suggestions on how to get the CLI program working. I guess I could add an optional `propagateBeets` option defaulting to no, and add it to top-level with that option enabled?

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Commit:
beets-bandcamp: init at 2022-06-07

necrophcodr commented:
> should i add myself to the maintainers list + here? (and should i do that in this PR?)
> 
> @lovesegfault would love a review in here if you have a chance + feedback on my idea to get the beetcamp cli working

According to the contributors guide it would be ideal to add yourself as a maintainer here, yes. You can do that in this PR too.

Commit:
add myself as maintainer to beets.beetcamp

Commit:
add myself to maintainers list

Commit:
optionally propagate beets package in beetsPackages.beetcamp

If beets is propagated while it is acting as a plugin, the parent
beets' files will not appear in the final package; if beets is not
propagated, beetcamp cli won't work.

This feels like an anti-pattern to me but I think it works? I'm not
sure if these packages will conflict in a system profile though.

onny commented:
Please rebase on top of master

arximboldi commented:
I see this is really old. Would love to have `beetcamp` included, what's needed to get this done?
